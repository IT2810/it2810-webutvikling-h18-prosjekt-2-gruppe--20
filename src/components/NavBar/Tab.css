
.tab {
  background-color: #dddddd;
  border: #aaaaaa solid 1px;
  box-sizing: border-box;

  color: #212121;

  margin: 0.5em;

  padding: 0.5em 1.5em;

  transition: all .2s ease;
}

.tab:hover {
  background-color: #eeeeee;
  box-shadow: 0.05em 0.1em 0.3em rgba(0,0,0,.05);
}

.tab:active {
  box-shadow: none;
}

.tab:focus {
  outline-width: 2px;
}

.tab--active {
  color: #aaaaaa;
  cursor: default;
}