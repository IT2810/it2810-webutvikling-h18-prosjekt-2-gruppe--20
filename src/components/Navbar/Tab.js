import React from 'react';
import PropTypes from 'prop-types';

import './Tab.css';

const Tab = ({
  children, id, onClick, isActive = false,
}) => {
  const clickHandler = () => onClick({ id });

  return <button
    className={`tab ${isActive ? 'tab--active' : ''}`}
    onClick={clickHandler}>{children}</button>;
};

Tab.propTypes = {
  id: PropTypes.any.isRequired,
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool,
};

export default Tab;
