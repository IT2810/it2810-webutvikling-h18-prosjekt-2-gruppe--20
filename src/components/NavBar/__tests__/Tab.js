import { render } from 'enzyme';

import Tab from '../Tab';

describe('<Tab/>', () => {
  it('should render default state', () => {
    const wrapper = render(<Tab id="hello" onClick={() => null}>Hello world</Tab>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render active state', () => {
    const wrapper = render(<Tab
      id="hello"
      onClick={() => null}
      isActive={true}>Hello selected world</Tab>);

    expect(wrapper).toMatchSnapshot();
  });
});
