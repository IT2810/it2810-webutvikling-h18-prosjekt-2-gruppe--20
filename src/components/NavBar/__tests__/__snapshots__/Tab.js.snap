// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab/> should render active state 1`] = `
<button
  class="tab tab--active"
>
  Hello selected world
</button>
`;

exports[`<Tab/> should render default state 1`] = `
<button
  class="tab "
>
  Hello world
</button>
`;
